src/test/java/pl/company/foodatu/FoodatuApplicationTests.java

```diff
@@ -373,12 +374,16 @@ void addMealToPlanForToday(String mealName, Map<UUID, Double> products) {
                         .map(product -> new ProductCreateDTO(product.getKey(), product.getValue()))
                         .toList()
         )).getBody();
-        try {
-            TimeUnit.SECONDS.sleep(1);
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
-        plansClient.addMealToPlan(new MealId(mealResponse.id()), USER.id(), TODAY);
+        Awaitility.await().atMost(3, TimeUnit.SECONDS)
+                .until(() -> {
+                    plansClient.addMealToPlan(new MealId(mealResponse.id()), USER.id(), TODAY);
+                    return true;
```

Comment:
So I guess you are not waiting here. In order to wait you should check if meal was added already

Reply:
The current code tries to add meal to plan for at most 3 seconds. If the meal is not available, the exception is thrown and it will try again. I tested it by adding delay (sleep) before publishing event and it worked. Should I still work on it?